iroh-bytes/src/protocol/range_spec.rs

```diff
@@ -1,32 +1,47 @@
-//! Specifications for individual ranges within a blob, and for ranges for a
-//! collection and its children.
+//! Specifications for ranges selection in blobs and collections.
+//!
+//! The [`RangeSpec`] allows specifying which BAO chunks inside a single blob should be
+//! selected.
+//!
+//! The [`RangeSpecSeq`] builds on top of this to select blob chunks in an entire
+//! collection.
 use std::fmt;
 
 use bao_tree::ChunkNum;
 use range_collections::{RangeSet2, RangeSetRef};
 use serde::{Deserialize, Serialize};
 use smallvec::{smallvec, SmallVec};
 
-/// A chunk range specification.
+/// A chunk range specification as a sequence of chunk offsets.
 ///
-/// this is a sequence of spans, where the first span is considered false, and each subsequent span is alternating.
+/// Offsets encode alternating spans starting on 0, where the first span is always deselected.
 ///
-/// Examples:
-/// The range 10..33 would be encoded as `[10, 23]`
-/// The empty range would be encoded as the empty array `[]`
-/// A full interval .. would be encoded as `[0]`
-/// A half open interval 15.. would be encoded as `[15]`
+/// ## Examples:
 ///
-/// All values except for the first one must be non-zero. The first value may be zero.
-/// Values are bao chunk numbers, not byte offsets.
+/// - `[2, 5, 3, 1]` encodes five spans, of which two are selected:
+///   - `[0, 0+2) = [0, 2)` is deselected.
```

Comment:
"not selected" instead of "deselected"? You either select something or you don't. Using "selected" and "deselected" for me sounds like there is two different things you can do, when there is really just one thing you can either do or not do.

Reply:
fixed